### PR TITLE
fix(staking miner): query latest state when comparing to already submitted solutions

### PR DESCRIPTION
**Pull request by niklasad1**

### CODE REVIEW

File: utils/staking-miner/src/monitor.rs [hunk 1]
```diff
@@ -138,6 +138,21 @@ async fn ensure_no_better_solution<T: EPM::Config, B: BlockT>(
...
 }
 
+async fn get_latest_head<T: EPM::Config>(
+	rpc: &SharedRpcClient,
+	mode: &str,
+) -> Result<Hash, Error<T>> {
+	if mode == "head" {
+		match rpc.block_hash(None).await {
+			Ok(Some(hash)) => Ok(hash),
+			Ok(None) => Err(Error::Other("Best head not found".into())),
```
Comment (niklasad1) [inline utils/staking-miner/src/monitor.rs:148]: this is silly, should be very very unlikely not to get the best block hash ^^

File: utils/staking-miner/src/monitor.rs [hunk 2]
```diff
@@ -138,6 +138,21 @@ async fn ensure_no_better_solution<T: EPM::Config, B: BlockT>(
 	}
 }
 
+async fn get_latest_head<T: EPM::Config>(
+	rpc: &SharedRpcClient,
+	mode: &str,
```
Comment (kianenigma) [inline utils/staking-miner/src/monitor.rs:143]: Should be enum? we can probably reuse the same enum in the CLI flags as well.

**Review by kianenigma — APPROVED** [review]
LGTM!

**Review by lexnv — APPROVED** [review]

**Review by jsdw — APPROVED** [review]
LGTM; I agree w.r.t using an enum instead of `"head"`

File: utils/staking-miner/src/monitor.rs (hunk 2 above)
Comment (niklasad1) [reply utils/staking-miner/src/monitor.rs:143]: haha, funny thing I just did that in the other miner repo :P

**Comment by niklasad1** [comment]
bot merge

**Pull request merged**